projects/igniteui-angular/src/lib/grids/grid-base.component.ts

```diff
@@ -2531,6 +2532,7 @@ export abstract class IgxGridBaseComponent extends DisplayDensityBase implements
         @Optional() @Inject(DisplayDensityToken) protected _displayDensityOptions: IDisplayDensityOptions) {
         super(_displayDensityOptions);
         this.resizeHandler = () => {
+            this.resetForOfCache();
```

Comment:
What about the `onColumnResized` event handler?

Reply:
I guess,  instead of trying to assign the top-most hor dir as master every time it would cause a cache or chunkSize change, we could have the service be informed by the grid that horizontal changes will be made, so that it will cause the update outside of the changeDetection cycle. It will require some refactoring inside the GridForOf code tho. I'll experiment with it a little.

Reply:
I guess, instead of trying to assign the top-most hor dir as master every time it would cause a cache or chunkSize change, we could have the service be informed by the grid that horizontal changes will be made, so that it will cause the update outside of the change detection cycle. It will require some refactoring inside the GridForOf code tho (e.g. the calcMaxChunkSize should be able to work of a parameter sent by the grid, not just the igxForOf input). I'll experiment with it a little.

Not sure I am fond of creating a "detection" patch like that but then again I am not fond of having to reset master on now 3 places inside the grid either. There is probably no win-win solution for this.

If the syncService wasn't giving the 10% rendering speed improvement I'd have removed it a month ago probably. It just doesn't play well with all the other perf optimization's we've made.

Reply:
> Is it possible to make the sync service inform the master cache (either through event/direct call) to recalculate the chunk size cache based on some grid interaction (resizing namely)?

I guess, instead of trying to assign the top-most hor dir as master every time it would cause a cache or chunkSize change, we could have the service be informed by the grid that horizontal changes will be made, so that it will cause the update outside of the change detection cycle. It will require some refactoring inside the GridForOf code tho (e.g. the calcMaxChunkSize should be able to work of a parameter sent by the grid, not just the igxForOf input). I'll experiment with it a little.

Not sure I am fond of creating a "detection" patch like that but then again I am not fond of having to reset master on now 3 places inside the grid either. There is probably no win-win solution for this.

If the syncService wasn't giving the 10% rendering speed improvement I'd have removed it a month ago probably. It just doesn't play well with all the other perf optimization's we've made.